### DISCUSSION TITLE
Can TypeSpec generate response.Content.ToString()?

### DISCUSSION THREAD

**Discussion by feyyangms** [microsoft/typespec#2557] [General]
Hello! I'm new to TypeSpec

For example, an endpoint defined in routes.tsp:
```
@route("/PetName")
@get
op getPetName(): {
  @statusCode statusCode: 200;
  @header contentType: "text/plain";
  @body body: string;
};
```

tspconfig.yaml (partial):
```
emit:
  - "@azure-tools/typespec-csharp"
```

Run `tsp compile .` to generate dotnet/C# client.

Result:
```
public virtual async Task<Response<string>> GetPetName(CancellationToken cancellationToken = default)
{
    RequestContext context = FromCancellationToken(cancellationToken);
    Response response = await GetPetName(context).ConfigureAwait(false);
    return Response.FromValue(response.Content.ToObjectFromJson<string>(), response);
}
```

The response body is a JSON string like "{"pet_name": "Meow"}"...but I want to keep it as a string instead of desacralize it into a Type...

That `ToObjectFromJson<string>()` is causing JSON deserialization error in this case.

Manually changing it to response.Content.ToString() worked - the JSON string from the response body is directly passed to `Response.FromValue(string, Response)`.

I'm assuming putting `@header contentType: "text/plain";` in the endpoint definition will tell TypeSpec to treat the response body as plain text instead of the default application/json, thus generating `response.Content.ToString()`. But seems that header specification isn't being picked up by TypeSpec?

Am I missing anything on what I want to achieve? Or should I report this as an issue? Appreciate your help!

**Comment by bterlson** [comment]
Your TypeSpec seems fine to me, I would expect pet_name on the wire as a bare string (without even any quotes). And indeed the [OpenAPI emit reflects this](https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwOwpAc2VydmljZSh7CiAgdGl0bGU6ICJXaWRnZXQgU8YbIiwKICB2ZXJzaW9uOiAiMS4wLjAiLAp9KQpuYW1lc3BhY2UgRGVtb8cvOwoKQHJvdXRlKCIvUGV0TmFtZSIpCkBnZXQKb3AgZ2V0xxUoKTogxHVAc3RhdHVzQ29kZSDKCzogMjAwO8QfaGVhZGVyIGNvbnRlbnTkAL86ICJ0ZXh0L3BsYWluIsUlYm9keSDEBTogc3RyaW5nOwp9Ow%3D%3D&e=%40typespec%2Fopenapi3&options=%7B%22linterRuleSet%22%3A%7B%22extends%22%3A%5B%22%40typespec%2Fhttp%2Fall%22%5D%7D%7D). This seems to be an issue with the code generator, maybe @m-nash can comment further.

**Comment by m-nash** [comment]
This is definitely a gap in our current generator.  I have opened this issue to track https://github.com/Azure/autorest.csharp/issues/3816

**Comment by pshao25** [comment]
@feyyangms This issue has been fixed. Please have a try.
